lib/asciidoctor/parser.rb

```diff
@@ -1234,7 +1235,10 @@ def self.catalog_inline_anchors text, document
     text.scan(InlineAnchorRx) do
       # lead with assignments for Ruby 1.8.7 compat
       id, reftext = $1 || $3, $2 || $4
-      document.register :ids, [id, reftext] unless $&.start_with? '\\'
+      unless $&.start_with? '\\'
+        document.register(:ids, [id, reftext])
+        document.register(:refs, [id, reftext])
```

Comment:
Shouldn't this save the block instead of the reftext?
The method should be called in Parser#next_block when a block is already initialized, so we can pass the block instead of document.